plugin-core/src/main/java/com/salesforce/bazel/eclipse/builder/BazelBuilder.java

```diff
@@ -86,20 +88,28 @@
 
     public static final String BUILDER_NAME = "com.salesforce.bazel.eclipse.builder";
 
+    private static final AtomicBoolean REGISTERED_EL_CHANGE_LISTENER = new AtomicBoolean(false);
     private static final LogHelper LOG = LogHelper.log(BazelBuilder.class);
 
+    public BazelBuilder() {
+        if (!REGISTERED_EL_CHANGE_LISTENER.getAndSet(true)) {
+            // is there a better way/place to register a singleton?
+            JavaCore.addElementChangedListener(new JDTWarningPublisher());
```

Comment:
@plaird let me know if you know of a real hook I can put this in.  I just want a single instance of JDTWarningPublisher registered.  I thought BazelNature may work, but nothing in there is called when Eclipse just restarts, with a bazel project already imported.  BazelBuilder is create for each project, that's why I added the boolean gate.

Reply:
BazelPluginActivator.startInternal is another candidate location.